src/hotspot/share/services/diagnosticFramework.hpp

```diff
@@ -440,12 +438,12 @@ template <class DCmdClass> class DCmdFactoryImpl : public DCmdFactory {
 
 private:
   template <typename T, ENABLE_IF(!std::is_base_of<DCmdWithParser, T>::value)>
-  static int get_num_arguments() {
+  static int get_parsed_num_arguments() {
     return T::num_arguments();
   }
 
   template <typename T, ENABLE_IF(std::is_base_of<DCmdWithParser, T>::value)>
-  static int get_num_arguments() {
+  static int get_parsed_num_arguments() {
```

Comment:
I don't understand why we need the dynamic allocation here. Should we not be able to call your new static versions of num_arguments like we do above? And if so, could we then not unify those two functions?

Reply:
Thanks for looking at this @tstuefe. The whole point of the parsed version is to ensure the static version doesn't get forgotten or out-of-date.

Just realised the `get_parsed_num_arguments` should also be `ifdef ASSERT`.